File: templates/go-service/service/httphandlers/helloPubSub.go
package httphandlers

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"registry.com/user/servicename/communication"
	"registry.com/user/servicename/context"
)

// CreateHelloPubSubHandler create helloPubSub handler
func CreateHelloPubSubHandler(context *context.Context, pubSub communication.PubSub) (handler func(c *gin.Context)) {
	return func(c *gin.Context) {
		// get name
		name := c.Param("name")
		if name == "" {
			name = c.Query("name")
		}
		if name == "" {
			name = c.PostForm("name")
		}
		if name == "" {
			name = "world"
		}

		// publish to DefaultRmqEvent of current service
		err := pubSub.Publish(context.Config.DefaultRmqEvent, communication.Message{"name": name})
		if err != nil {
			context.Config.Logger.Println(err)
			c.String(http.StatusInternalServerError, "Sending error")
			return
		}
		c.String(http.StatusOK, "Message sent")

	}
}

File: templates/nodejs-service/service/httphandlers/helloPubSub.js
function createHelloPubSubHandler(context, pubSub) {
    return function hello(req, res) {
        const config = context.getConfig();
        let name = req.params.name;
        if (!name) {
            name = req.query.name;
        }
        if (!name) {
            name = req.body.name;
        }
        if (!name) {
            name = "world";
        }
        try {
            pubSub.publish(config.defaultRmqEvent, { name });
        } catch (err) {
            config.logger.error(err);
            return res.status(500).send("Sending error");
        }
        res.send("Message sent");
    }
}

module.exports = { createHelloPubSubHandler };